python/ray/autoscaler/_private/kuberay/autoscaling_config.py

```diff
@@ -262,6 +263,33 @@ def _get_ray_resources_from_group_spec(
     if num_gpus is not None:
         resources["GPU"] = num_gpus
 
+    if num_tpus is not None:
```

Comment:
Is this check necessary? If we implement it, conducting autoscaling end-to-end tests without TPU machines becomes challenging. Specifically, we do not set the TPU in the Pod resources, but we pass the TPU to `rayStartParams` to deceive Ray into autoscaling as if it were a multi-host TPU.

Reply:
I'm not sure I understand, in the case where a user passes TPUs to `rayStartParams`, wouldn't they expect the autoscaler to attempt to scale as though there were TPUs (even if it's not able to due to the k8s resource limits not being set correctly)? We could fix this by changing `_get_num_tpus` to only check whether TPUs are requested in the Pod resources, I had just had it checking `rayStartParams` to match the logic already implemented for cpus and gpus.

Reply:
I think the right behavior here is to use what's available in the k8s pod resources.

Reply:
Done, I changed `_get_num_tpus` to only check the k8s pod resources and add the head resource if TPUs are detected there.

Reply:
Oh then the previous code matched the desired behavior. Previously, I had `_get_num_tpus` checking first `rayStartParams` for a TPU custom resource, and then `k8s_resource_limits` for `google.com/tpu` limits, the same way as `_get_num_gpus`. This would enable faking multi-host TPU podslices in tests, since you could just pass in a TPU value to `rayStartParams: resources:`, allowing the head resource to get added, without setting the k8s Pod limits. I'll change the code back to match this behavior.